src/rust/ensogl/lib/components/src/list_view/entry.rs

```diff
@@ -18,148 +20,261 @@ use ensogl_theme;
+    /// in [`text::Area::add_to_scene_layer`] method.
+    fn set_label_layer(&self, label_layer:&display::scene::Layer);
 }
 
-impl Model {
-    /// Create model of simple entry with given label.
-    ///
-    /// The model won't have icon nor higlighting, but those can be set using `highlight` and
-    /// `with_icon`.
-    pub fn new(label:impl Str) -> Self {
-        Self {
-            label       : label.into(),
-            highlighted : default(),
-            icon        : default(),
+
+// =======================
+// === Implementations ===
+// =======================
+
+// === Label ===
+
+/// The [`Entry`] being a single text field displaying String.
+#[derive(Clone,CloneRef,Debug)]
+pub struct Label {
+    display_object : display::object::Instance,
+    label          : text::Area,
+    network        : enso_frp::Network,
+    style_watch    : StyleWatchFrp,
+}
+
+impl Entry for Label {
+    type Model = String;
+
+    fn new(app: &Application) -> Self {
+        let logger         = Logger::new("list_view::entry::Label");
+        let display_object = display::object::Instance::new(logger);
+        let label          = app.new_view::<ensogl_text::Area>();
+        let network        = frp::Network::new("list_view::entry::Label");
+        let style_watch    = StyleWatchFrp::new(&app.display.scene().style_sheet);
+        let color          = style_watch.get_color(theme::widget::list_view::text);
+        let size           = style_watch.get_number(theme::widget::list_view::text::size);
+
+        display_object.add_child(&label);
+        frp::extend! { network
+            init <- source::<()>();
+            color <- all(&color,&init)._0();
+            size  <- all(&size,&init)._0();
+
+            label.set_default_color     <+ color;
+            label.set_default_text_size <+ size.map(|v| text::Size(*v));
+            eval size ((size) label.set_position_y(size/2.0));
         }
+        init.emit(());
+        Self {display_object,label,network,style_watch}
     }
 
-    /// Add highlighting to the entry and return it.
-    pub fn highlight(mut self, bytes:impl IntoIterator<Item=text::Range<text::Bytes>>) -> Self {
-        self.highlighted.extend(bytes.into_iter());
-        self
+    fn update(&self, model: &Self::Model) {
+        self.label.set_content(model);
     }
 
-    /// Add icon to the entry and return it.
-    pub fn with_icon(mut self, icon:impl display::Object + 'static) -> Self {
-        self.icon = Some(icon.into_any());
-        self
+    fn set_label_layer(&self, label_layer:&display::scene::Layer) {
+        self.label.add_to_scene_layer(label_layer);
     }
 }
 
-impl<T:Display> From<T> for Model {
-    fn from(item: T) -> Self {
-        Model::new(item.to_string())
+impl display::Object for Label {
+    fn display_object(&self) -> &display::object::Instance { &self.display_object }
+}
+
+
+// === HighlightedLabel ===
+
+/// The model for [`HighlightedLabel`]
+#[derive(Clone,Debug,Default)]
```

Comment:
These docs should describe hwat it is. Until I went deeply into the code I was thinking its a bout a big box-highlighting in the backgorund

Reply:
Also maybe changing the name to GlyphHighlightedLabel would make it working better?